lib/_tls_wrap.js

```diff
@@ -240,6 +245,46 @@ function onnewsession(key, session) {
     done();
 }
 
+function onPskServerCallback(identity, maxPskLen) {
+  const owner = this[owner_symbol];
+  const psk = owner[kPskCallback](identity);
+  if (psk == null) {
+    return undefined;
+  } else {
+    if (typeof psk !== 'string') {
+      throw new ERR_INVALID_ARG_TYPE('psk', ['string']);
+    }
+    if (psk.length / 2 > maxPskLen) {
+      throw new ERR_ASSERTION(`Pre-shared key exceeds ${maxPskLen} bytes`);
+    }
+    return psk;
+  }
+}
+
+function onPskClientCallback(hint, maxPskLen, maxIdentityLen) {
```

Comment:
how dynamic is this max? we don't expose it to the API user (other than throwing an error that they can't handle). That's OK if its larger than all reasonable identity values.

Reply:
Actually it's not that big, according to the openssl sources
PSK - 256 bytes
Identity 128 bytes
```cpp
deps/openssl/openssl/include/openssl/ssl.h
722:#  define PSK_MAX_IDENTITY_LEN 128
723:#  define PSK_MAX_PSK_LEN 256
```
IIUC.

Though what can we do here?